Documentation/Patterns.md

```diff
@@ -280,3 +280,12 @@ private:
 };
 #endif
 ```
+
+## `type[]` vs. `Array` vs. `Vector` vs. `FixedArray`
+
+There are four "contiguous list" / array-like types, including C-style arrays themselves. They share a lot of their API, but their use cases are all slightly different, mostly relating to how they allocate their data.
+
+* C-style arrays are generally discouraged (and this also holds for pointer+size-style arrays when passing them around). They are only used for the implementation of other collections or in specific circumstances.
+* `Array` is a thin wrapper around C-style arrays, where the template arguments include the size of the array. It allocates its data inline, just as arrays do, and never does any dynamic allocations.
+* `Vector` is similar to `std::vector` and represents a dynamic resizeable array. For most basic use cases of lists, this is the go-to collection. It has an optional inline capacity (the second template argument) which will allocate inline as the name suggests, but this is not always used. If the inline capacity becomes too small, Vector will automatically switch to the standard out-of-line storage. This is allocated on the heap, and the space is automatically resized and moved when more (or less) space is needed.
```

Comment:
"If the inline capacity becomes too small" makes it sound more like the capacity shrinks. I would suggest changing it to "If the contents outgrow the inline capacity", or something similar.